opm/simulators/utils/readDeck.cpp

```diff
@@ -557,12 +557,16 @@ void Opm::readDeck(Opm::Parallel::Communication    comm,
             auto parseContext = setupParseContext(exitOnAllErrors);
             if (treatCriticalAsNonCritical) { // Continue with invalid names if parsing strictness is set to low
                 parseContext->update(ParseContext::SCHEDULE_INVALID_NAME, InputErrorAction::WARN);
-                schedule->treat_critical_as_non_critical(true);
             }
             readOnIORank(comm, deckFilename, parseContext.get(),
                          eclipseState, schedule, udqState, actionState, wtestState,
                          summaryConfig, std::move(python), initFromRestart,
                          checkDeck, treatCriticalAsNonCritical, outputInterval, *errorGuard);
+
+            if (treatCriticalAsNonCritical) { // Update schedule so that re-parsing after actions use same strictness
+                assert(schedule);
+                schedule->treat_critical_as_non_critical(true);
+            }
```

Comment:
I guess you could, strictly speaking, write this as
```C++
schedule->treat_critical_as_non_critical(treatCriticalAsNonCritical);
```
instead.  It's not clear to me if that's easier to read though.

Reply:
Ah, I think that's slightly better. Thanks, will update.